tests/integ/test_transformer.py

```diff
@@ -104,6 +104,46 @@ def test_attach_transform_kmeans(sagemaker_session):
         attached_transformer.wait()
 
 
+def test_transformer_delete_model(sagemaker_session):
```

Comment:
I think we can maybe make use of the [timeout](https://github.com/aws/sagemaker-python-sdk/blob/7f33ac79cd4b10f58eafd7a1b2f9af0dc5c48edd/tests/integ/timeout.py) class for handling delete model.

Otherwise, I wouldn't advise going through the training and transform in a whole new test in order to test the delete_model method, since you are only testing deleting the corresponding model and don't necessarily care about the training or deploy portion, since this is the transformer and we can assume they have to come in with a SageMaker model. I think it is okay to use an existing test and modify it to include deleting of the model.

Either that, or create a test that reuses the model of another test.

Reply:
You can replace the timeout with your time function. Place the assertion within the timeout.

Reply:
This test is deleted because I replaced the timeout with the updated timeout function.